src/rdf4cpp/writer/BufWriter.hpp

```diff
@@ -79,18 +52,20 @@ concept BufWriter = requires (W &bw, size_t additional_cap) {
     /**
      * Flushes bw.buffer() and repoints bw.cursor() to the new free space
      */
-    W::flush(&bw.buffer(), &bw.cursor(), additional_cap);
+    W::flush(&bw.buffer(), &bw.write_area(), &bw.write_area_size(), additional_cap);
 };
 
 /**
  * Flushes user defined data.
- * the functions task is it to somehow make room in buffer and point the cursor to that new, free space
+ * the functions task is it to somehow make room in buffer and point the cursor and remaining_size to that new, free space
  *
  * @param buffer buffer to make room in
- * @param cursor a pointer + size pair into buffer (pointing to the free space)
+ * @param write_area a pointer to the start of the free space, should be moved to the start of the new free space
+ * @param write_area_size the remaining size of the free space, should be set to the size of the new free space
  * @param additional_cap how much additional space is needed right now
+ * @note should not throw exceptions, but is not marked noexcept so that C functions can be easily used
```

Comment:
```suggestion
 * @note must not throw exceptions, but is not marked noexcept so that C functions can be easily used
```

not entirely sure about this one